src/octree.cpp

```diff
@@ -40,165 +40,200 @@ namespace fcl
+    vertices.push_back(Vec3f(x+.5*size, y-.5*size, z+.5*size));
+    vertices.push_back(Vec3f(x-.5*size, y+.5*size, z+.5*size));
+    vertices.push_back(Vec3f(x+.5*size, y+.5*size, z+.5*size));
+    
+    // Add face only if box has no neighbor with the same face
+    if (!n.minusX())
+    {
+      Array4 a = {{8*i + 1, 8*i + 5, 8*i + 7, 8*i + 3}};
+      faces.push_back(a);
     }
-    // write obj in a file
-    std::ofstream os;
-    os.open(filename);
-    if (!os.is_open())
-      throw std::runtime_error(std::string("failed to open file \"")+
-			       filename + std::string("\""));
-      // write vertices
-    os << "# list of vertices" << std::endl;
-    for (auto v : vertices){
-      os << "v " << v[0] << " " << v[1] << " " << v[2] << std::endl;
+    if (!n.plusX())
+    {
+      Array4 a = {{8*i + 2, 8*i + 4, 8*i + 8, 8*i + 6}};
+      faces.push_back(a);
     }
-    os << std::endl << "# list of faces" << std::endl;
-    for (auto f : faces) {
-      os << "f " << f[0] << " " << f[1] << " " << f[2] << " " << f[3]
-	 << std::endl;
+    if (!n.minusY())
+    {
+      Array4 a = {{8*i + 1, 8*i + 2, 8*i + 6, 8*i + 5}};
+      faces.push_back(a);
     }
-  }
-
-  OcTreePtr_t makeOctree
-  (const Eigen::Matrix<FCL_REAL,Eigen::Dynamic,3>& point_cloud,
-   const FCL_REAL & resolution)
-  {
-    typedef Eigen::Matrix<FCL_REAL,Eigen::Dynamic,3> InputType;
-    typedef InputType::ConstRowXpr RowType;
-    
-    boost::shared_ptr<octomap::OcTree> octree(new octomap::OcTree(resolution));
-    for(Eigen::DenseIndex row_id = 0; row_id < point_cloud.rows(); ++row_id)
+    if (!n.plusY())
     {
-      RowType row = point_cloud.row(row_id);
-      octomap::point3d p(static_cast<float>(row[0]),static_cast<float>(row[1]),static_cast<float>(row[2]));
-      octree->updateNode(p, true);
+      Array4 a = {{8*i + 4, 8*i + 3, 8*i + 7, 8*i + 8}};
+      faces.push_back(a);
+    }
+    if (!n.minusZ())
+    {
+      Array4 a = {{8*i + 1, 8*i + 2, 8*i + 4, 8*i + 3}};
+      faces.push_back(a);
+    }
+    if (!n.plusZ())
+    {
+      Array4 a = {{8*i + 5, 8*i + 6, 8*i + 8, 8*i + 7}};
+      faces.push_back(a);
     }
-    octree->updateInnerOccupancy();
     
-    return OcTreePtr_t (new OcTree(octree));
   }
+  // write obj in a file
+  std::ofstream os;
+  os.open(filename);
+  if (!os.is_open())
+    throw std::runtime_error(std::string("failed to open file \"")+
+                             filename + std::string("\""));
+  // write vertices
+  os << "# list of vertices" << std::endl;
+  for (VectorVec3f::const_iterator it = vertices.begin();
+       it != vertices.end(); ++it)
+  {
+    const Vec3f & v = *it;
+    os << "v " << v[0] << " " << v[1] << " " << v[2] << std::endl;
+  }
+  os << std::endl << "# list of faces" << std::endl;
+  for (VectorArray4::const_iterator it = faces.begin();
+       it != faces.end(); ++it)
+  {
+    const Array4 & f = *it;
+    os << "f " << f[0] << " " << f[1] << " " << f[2] << " " << f[3] << std::endl;
+  }
```

Comment:
I know this is not the scope of this PR but, for large octrees, the `std::endl` will be very inefficient.

```suggestion
  os << "# list of vertices\n";
  for (VectorVec3f::const_iterator it = vertices.begin();
       it != vertices.end(); ++it)
  {
    const Vec3f & v = *it;
    os << "v " << v[0] << " " << v[1] << " " << v[2] << '\n';
  }
  os << "\n# list of faces\n";
  for (VectorArray4::const_iterator it = faces.begin();
       it != faces.end(); ++it)
  {
    const Array4 & f = *it;
    os << "f " << f[0] << " " << f[1] << " " << f[2] << " " << f[3] << '\n';
  }
```